airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/CdcMssqlSourceAcceptanceTest.java

```diff
@@ -0,0 +1,240 @@
+        "com.microsoft.sqlserver.jdbc.SQLServerDriver",
+        null);
+
+    config = Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", container.getHost())
+        .put("port", container.getFirstMappedPort())
+        .put("database", DB_NAME)
+        .put("username", TEST_USER_NAME)
+        .put("password", TEST_USER_PASSWORD)
+        .put("replication_method", "CDC")
+        .build());
+
+    executeQuery("CREATE DATABASE " + DB_NAME + ";");
+    executeQuery("ALTER DATABASE " + DB_NAME + "\n\tSET ALLOW_SNAPSHOT_ISOLATION ON");
+    executeQuery("USE " + DB_NAME + "\n" + "EXEC sys.sp_cdc_enable_db");
+
+    setupTestUser();
+    revokeAllPermissions();
+    createAndPopulateTables();
+    grantCorrectPermissions();
+  }
+
+  private void setupTestUser() {
+    executeQuery("USE " + DB_NAME);
+    executeQuery("CREATE LOGIN " + TEST_USER_NAME + " WITH PASSWORD = '" + TEST_USER_PASSWORD + "';");
+    executeQuery("CREATE USER " + TEST_USER_NAME + " FOR LOGIN " + TEST_USER_NAME + ";");
+  }
+
+  private void revokeAllPermissions() {
+    executeQuery("REVOKE ALL FROM " + TEST_USER_NAME + " CASCADE;");
+    executeQuery("EXEC sp_msforeachtable \"REVOKE ALL ON '?' TO " + TEST_USER_NAME + ";\"");
+  }
+
+  private void createAndPopulateTables() throws InterruptedException {
+    executeQuery(String.format("CREATE TABLE %s.%s(id INTEGER PRIMARY KEY, name VARCHAR(200));",
+        SCHEMA_NAME, STREAM_NAME));
+    executeQuery(String.format("INSERT INTO %s.%s (id, name) VALUES (1,'picard'),  (2, 'crusher'), (3, 'vash');",
+        SCHEMA_NAME, STREAM_NAME));
+    executeQuery(String.format("CREATE TABLE %s.%s(id INTEGER PRIMARY KEY, name VARCHAR(200));",
+        SCHEMA_NAME, STREAM_NAME2));
+    executeQuery(String.format("INSERT INTO %s.%s (id, name) VALUES (1,'enterprise-d'),  (2, 'defiant'), (3, 'yamato');",
+        SCHEMA_NAME, STREAM_NAME2));
+
+    // sometimes seeing an error that we can't enable cdc on a table while sql server agent is still
+    // spinning up
+    // solving with a simple while retry loop
+    boolean failingToStart = true;
+    int retryNum = 0;
+    int maxRetries = 10;
+    while (failingToStart) {
+      try {
+        // enabling CDC on each table
+        String[] tables = {STREAM_NAME, STREAM_NAME2};
+        for (String table : tables) {
+          executeQuery(String.format(
+              "EXEC sys.sp_cdc_enable_table\n"
+                  + "\t@source_schema = N'%s',\n"
+                  + "\t@source_name   = N'%s', \n"
+                  + "\t@role_name     = N'%s',\n"
+                  + "\t@supports_net_changes = 0",
+              SCHEMA_NAME, table, CDC_ROLE_NAME));
+        }
+        failingToStart = false;
+      } catch (Exception e) {
+        if (retryNum >= maxRetries) {
+          throw e;
+        } else {
+          retryNum++;
+          Thread.sleep(10000); // 10 seconds
+        }
+      }
+    }
+  }
+
+  private Boolean isSqlServerAgentRunning() {
```

Comment:
I think this method is now unused (on me, I left it there after removing use of it)